File: src/main/java/net/yogstation/yogbot/http/byond/IByondEndpoint.java
package net.yogstation.yogbot.http.byond;

import com.fasterxml.jackson.databind.JsonNode;
import org.springframework.http.HttpEntity;
import reactor.core.publisher.Mono;

public interface IByondEndpoint {
	String getMethod();
	Mono<HttpEntity<String>> receiveData(JsonNode data);
}

File: src/main/java/net/yogstation/yogbot/http/byond/IByondEndpoint.kt
package net.yogstation.yogbot.http.byond

import com.fasterxml.jackson.databind.JsonNode
import org.springframework.http.HttpEntity
import reactor.core.publisher.Mono

interface IByondEndpoint {
	val method: String
	fun receiveData(data: JsonNode): Mono<HttpEntity<String>>
}
